tests/maintel/test_mtcs.py

```diff
@@ -1978,3 +1980,79 @@ def assert_m1m3_booster_valve_no_m1m3(self) -> None:
             self.mtcs.rem.mtm1m3.evt_boosterValveStatus.next.assert_not_awaited()
             self.mtcs.rem.mtm1m3.cmd_boosterValveOpen.start.assert_not_awaited()
             self.mtcs.rem.mtm1m3.cmd_boosterValveClose.start.assert_not_awaited()
+
+    async def test_set_m1m3_slew_controller_settings_changes_setting(self) -> None:
+        # Test changing a setting
+        await self.run_set_m1m3_slew_controller_setting_test(
+            initial_setting_value=False,
+            desired_setting=MTM1M3.SetSlewControllerSettings.ACCELERATIONFORCES,
+            desired_value=True,
+        )
+
+    async def test_set_m1m3_slew_controller_settings_no_change_for_same_setting(
+        self,
+    ) -> None:
+        # Test no change for the same setting
+        await self.run_set_m1m3_slew_controller_setting_test(
+            initial_setting_value=True,
+            desired_setting=MTM1M3.SetSlewControllerSettings.ACCELERATIONFORCES,
+            desired_value=True,
+        )
+
+    async def run_set_m1m3_slew_controller_setting_test(
+        self,
+        initial_setting_value: bool,
+        desired_setting: MTM1M3.SetSlewControllerSettings,
+        desired_value: bool,
+    ) -> None:
+        # Prepare initial settings
```

Comment:
Consider moving these things to the `MTCSAsyncMock` class. and actually implementing a mock to the internals of `get_m1m3_slew_controller_settings`, instead of mocking the entire thing. 


Basically, in [setup_types](https://github.com/lsst-ts/ts_observatory_control/blob/a5f61067fbb3ab3fe14bc44d33e0f3420dd6fb32/python/lsst/ts/observatory/control/mock/mtcs_async_mock.py#L78C15-L78C26) you should set an attribute

```
self._evt_slew_controller_settings = types.SimpleNamespace(
            useAccelerationForces=False,
            useBalanceForces=False,
            triggerBoosterValves=False,
            useVelocityForces=False,
)
```

then define a method to retrieve the value, `mtm1m3_evt_slew_controller_settings`:

```
async def mtm1m3_evt_slew_controller_settings(
        self, *args: typing.Any, **kwargs: typing.Any
    ) -> types.SimpleNamespace:
        await asyncio.sleep(self.heartbeat_time / 4.0)
        return self._evt_slew_controller_settings
```

and set it up in the `setup_mtm1m3` method:

```
m1m3_mocks = {
          ...,
          "evt_slewControllerSettings.aget.side_effect": mtm1m3_evt_slew_controller_settings,
          ...
}
```

Then do a similar thing on the command side to mock the behavior of setting the value.

Reply:
I have it working, but not happy with everything. Please take a look and tell me if you see anything wrong or if you see things to be improved.